### ISSUE TITLE
DualShock not supported for early PS1 Games

### ISSUE THREAD

**Issue by Catarax** [libretro/beetle-psx-libretro#83]
If you play a game like Dragon Ball GT Final Bout with a Dualshock, you'll not be able to play the game :

![56d4c65256f640bd970ce28685cd1d8e](https://cloud.githubusercontent.com/assets/3964507/17979445/e301d598-6afa-11e6-82bd-8b7af105b1ca.png)
Along with Tekken 2, Rayman 1, Disney Hercules...

That's normal, ok but I would like you to not having to switch everytime just for having the rumble feature.

I've reported this error for the LilyPad plugin of PCSX2 back in the days and they were able to fix the issue in order to play the game despite the dualshock option activated.
(https://github.com/PCSX2/pcsx2/issues/1348)

The line of code they used:
`if (pad->mode == MODE_DIGITAL && config.padConfigs[query.port][query.slot].autoAnalog && !ps2e) {pad->mode = MODE_ANALOG;}`

Hope you can do the same for Beetle-psx, thanks!

**Comment by simias** [comment]
Don't you need a game database to know whether or not it supports DualShock? I can't think of an other way...

**Comment by Catarax** [comment]
I'm asking a better support for DualShock on early games because they have trouble to run due to their analog feature, that's all. This is fixable since PCSX2 was able to fix their plugin. I don't need a database, the game itself will tell me that I can't play at startup or else the controller will be useless during the game.

**Comment by Papermanzero** [comment]
This is however an accurate emulation. If tou need the classic conteoller switch in the controller options to classic controller.

**Comment by Catarax** [comment]
Yes sure but original mednafen switch automatically for games like that, you don't have to bother.
Anyway I switched over this one one since.

**Comment by Rican7** [comment]
> Don't you need a game database to know whether or not it supports DualShock? I can't think of an other way...

So after a lot of my own research with this, I realized that libretro itself actually has such a database:

 - [PlayStation (PS1/PSX) **analog** support metadata (list)](https://github.com/libretro/libretro-database/blob/v1.9.0/metadat/analog/Sony%20-%20PlayStation.dat)
 - [PlayStation (PS1/PSX) **rumble** support metadata (list)](https://github.com/libretro/libretro-database/blob/v1.9.0/metadat/rumble/Sony%20-%20PlayStation.dat)

**Comment by Rican7** [comment]
... in which case, would it be possible to somehow query for this database metadata upon the loading of a game/content and default certain options based on it?
